spec/lib/api/v3/custom_fields/hierarchy/hierarchy_item_representer_rendering_spec.rb

```diff
@@ -0,0 +1,223 @@
+
+  context "if item is root" do
+    let(:item) { root }
+
+    describe "_links" do
+      it_behaves_like "has an untitled link" do
+        let(:link) { "self" }
+        let(:href) { api_v3_paths.custom_field_item(item.id) }
+      end
+
+      it_behaves_like "has no link" do
+        let(:link) { "parent" }
+      end
+
+      it_behaves_like "has a link collection" do
+        let(:link) { "children" }
+        let(:hrefs) do
+          [
+            {
+              href: api_v3_paths.custom_field_item(luke.id),
+              title: luke.label
+            },
+            {
+              href: api_v3_paths.custom_field_item(mara.id),
+              title: mara.label
+            }
+          ]
+        end
+      end
+
+      it_behaves_like "has an untitled link" do
+        let(:link) { "branch" }
+        let(:href) { api_v3_paths.custom_field_item_branch(item.id) }
+      end
+    end
+
+    describe "properties" do
+      it_behaves_like "property", :_type do
+        let(:value) { "HierarchyItem" }
+      end
+
+      it_behaves_like "property", :id do
+        let(:value) { item.id }
+      end
+
+      it_behaves_like "property", :label do
+        let(:value) { nil }
+      end
+
+      it_behaves_like "property", :short do
+        let(:value) { nil }
+      end
+
+      it_behaves_like "property", :depth do
+        let(:value) { 0 }
+      end
+    end
+
+    context "and depth is negative" do
+      let(:item) { API::V3::CustomFields::Hierarchy::HierarchicalItemAggregate.new(item: root, depth: -1) }
+
+      describe "properties" do
+        it_behaves_like "property", :depth do
+          let(:value) { nil }
+        end
+      end
+    end
+  end
+
+  context "if item is leave" do
+    let(:item) { mouse }
+
+    describe "_links" do
+      it_behaves_like "has a titled link" do
+        let(:link) { "self" }
+        let(:href) { api_v3_paths.custom_field_item(item.id) }
+        let(:title) { item.label }
+      end
+
+      it_behaves_like "has a titled link" do
+        let(:link) { "parent" }
+        let(:href) { api_v3_paths.custom_field_item(r2d2.id) }
+        let(:title) { r2d2.label }
+      end
+
+      it_behaves_like "has a link collection" do
+        let(:link) { "children" }
+        let(:hrefs) { [] }
+      end
+
+      it_behaves_like "has an untitled link" do
+        let(:link) { "branch" }
+        let(:href) { api_v3_paths.custom_field_item_branch(item.id) }
+      end
+    end
+
+    describe "properties" do
+      it_behaves_like "property", :_type do
+        let(:value) { "HierarchyItem" }
+      end
+
+      it_behaves_like "property", :id do
+        let(:value) { item.id }
+      end
+
+      it_behaves_like "property", :label do
+        let(:value) { item.label }
+      end
+
+      it_behaves_like "property", :short do
+        let(:value) { item.short }
+      end
+
+      it_behaves_like "property", :depth do
+        let(:value) { item.depth }
```

Comment:
🔴 moment, why does this work?? `item.depth` would execute the the depth query of an `HierarchicalItem`, it should not be the same as the rendered value in the JSON?